### PR TITLE
feat: pallet proxy

### PR DESCRIPTION
**Pull request by al3mart**
> :warning: This needs #68 & #69 to be merged first!

Adds `pallet-proxy` to Pop Network runtimes:
- [x] devnet
- [x] testnet

### CODE REVIEW

**Comment by evilrobot-01** [comment]
Rebased after other two listed PRs merged.

File: runtime/devnet/src/lib.rs [hunk 1]
```diff
@@ -9,6 +9,7 @@ include!(concat!(env!("OUT_DIR"), "/wasm_binary.rs"));
 mod assets_config;
 mod contracts_config;
 mod extensions;
+mod proxy_config;
```
[the remark this answers is not in the document]
Comment (al3mart) [reply runtime/devnet/src/lib.rs:12]: That makes sense!

[the remark this answers is not in the document]
Comment (al3mart) [reply runtime/devnet/src/lib.rs:12]: Let's maybe address this in a different pr @evilrobot-01 and get this merged

[the remark this answers is not in the document]
Comment (evilrobot-01) [reply runtime/devnet/src/lib.rs:12]: The issue when running with `runtime-benchmarks `fundamentally stems from the workaround encountered at https://github.com/r0gue-io/pop-node/pull/48/commits/4f337e46962b080b7c48e6a03faffe08c76bc8d6. 

I think the only easy/real solution would be to update to 1.8.

[the remark this answers is not in the document]
Comment (evilrobot-01) [reply runtime/devnet/src/lib.rs:12]: Looking at the other working branch, I agree that it should be dealt with in a separate PR. Is this good to convert to ready to review now?

**Review by Daanvdplas — APPROVED** [review]

**Review by peterwht — APPROVED** [review]

**Comment by evilrobot-01** [comment]
This PR introduces ~150 lines of duplicate code, signalling future technical debt. Appreciate not all of it can be refactored away due to different runtime types, but surely at least the `parameter_types` and `ProxyType` can be moved to common?

Its easy to just duplicate things, but the burden of later maintenance should be considered.

**Comment by al3mart** [comment]
I am happy to address this.

@evilrobot-01 based on previous comments my understanding was that most likely we would move away from maintaining different runtimes. Though this should not be a excuse for not having a proper code base today.

**Review by evilrobot-01 — APPROVED** [review]
LGTM - pushed one final commit to further eliminate duplicate code.

**Review by peterwht — APPROVED** [review]

**Pull request merged**